experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,241 @@
+  charTable=character_table_complex_reflection_group(r,1,n)
+  charTirr=[charTable[i,j] for i in 1:nrows(charTable), j in 1:ncols(charTable)]
+  Modules=[multipartition([lbb...]) for lbb in class_parameters(charTable)]
+  l=length(Modules)
+  charTirr=reorder(charTirr,Modules,mps)
+  charTirrT=solve_init(matrix(Q,transpose(charTirr)))
+  rows=zero_matrix(Q,l,0)
+  for i in 1:l
+    col=zero_matrix(Q,l,1)
+    for j in 1:l
+      v=matrix(Q,l,1,map(k->charTirr[i,k]*charTirr[j,k],1:l))
+      x=solve(charTirrT,v,side=:right)
+      col=col+fake_deg(mps[j],Q,var)*x
+    end
+    rows=hcat(rows,col)
+  end
+  return rows
+end
+
+# computes the character of the simple representations of the Cherednik algebra.
+function C_L(r::Int, n::Int, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}}, var::AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}, mps::Vector{Multipartition{Int}})
+  C_D=C_Delta(r,n,Q,var,mps)
+  d = [var^b_inv(mp)//fake_deg(mp,Q,var) for mp in mps]
+  diag=diagonal_matrix(Q,d)
+  return diag * C_D
+end
+
+function bigger_ord(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int}, mps::Vector{Multipartition{Int}})
+  r=length(lbb)
+  n=sum(lbb)
+  lbbquot=tau_om(lbb,wperm,coroot)
+  res=map(x->x,Iterators.filter(x-> dominates(tau_om(x,wperm,coroot),lbbquot),mps))
+  return res
+end
+
+function smaller_ord(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int}, mps::Vector{Multipartition{Int}})
+  r=length(lbb)
+  n=sum(lbb)
+  lbbquot=tau_om(lbb,wperm,coroot)
+  res=map(x->x,Iterators.filter(x-> dominates(lbbquot,tau_om(x,wperm,coroot)),mps))
+  return res
+end
+
+@doc raw"""
+    wreath_macdonald_polynomials(n::Int,
+                                 r::Int,
+                                 wperm::PermGroupElem,
+                                 coroot::Vector{Int};
+                                 parent::AbstractAlgebra.Generic.MPolyRing{QQAbFieldElem{AbsSimpleNumFieldElem}})
+
+Given two integers n and r and an element of the affine Weyl group of type A (seen as
+the semi-direct product of the Symmetric group with the coroot lattice), this function
+returns the square matrix of coefficients of the wreath Macdonald polynomials associated with
+all multipartition of size n and length r in the standard Schur basis indexed by the
+multipartitions of the same size and length as lbb. Each row of this matrix is a wreath
+Macdonald polynomial.
+"""
+function wreath_macdonald_polynomials(n::Int,
+                                      r::Int,
+                                      wperm::PermGroupElem,
+                                      coroot::Vector{Int};
+                                      parent::AbstractAlgebra.Generic.MPolyRing{QQAbFieldElem{AbsSimpleNumFieldElem}})
```

Comment:
Let's provide a default `parent` for the user's convenience:
```suggestion
                                      parent::MPolyRing{<:QQAbFieldElem} = polynomial_ring(K,[:q,:t];cached=false))
```
Actually I think we have allowed this to be a cached ring for added convenience in other cases? @thofma ?
```suggestion
                                      parent::MPolyRing{<:QQAbFieldElem} = polynomial_ring(K,[:q,:t];cached=true))
```

Reply:
Isn't `polynomial_ring` returning a tuple?
By default I can cache the ring if that is more convenient.

Reply:
I changed it to `parent::MPolyRing{<:QQAbFieldElem} = polynomial_ring(abelian_closure(QQ)[1],[:q,:t];cached=true)[1])`.